### PR TITLE
Feature: migrate import gait files

### PR DESCRIPTION
**Pull request by Roelemans**
Dependent on https://github.com/project-march/march/pull/404

Closes PM-148

## Description

Remove the import statements and use the constructors defined in https://github.com/project-march/march/pull/404. 

## Changes

* Removed import function
* Removed gait factory
* Create emtpy_subgait method in ModifableSubgait
* Override from_msg method in ModifiableJointTrajectory to use user-defined setpoints instead of trajectory points.

### CODE REVIEW

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py [hunk 1]
```diff
@@ -23,6 +23,39 @@ def __init__(self, name, limits, setpoints, duration, gait_generator):
 
         self.interpolated_setpoints = self.interpolate_setpoints()
 
+    @classmethod
+    def from_msg(cls, subgait_msg, joint_name, limits, duration):
```
[the remark this answers is not in the document]
Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py]: That happens within the 'from_dict' method within the subgait class before this method is called, see https://github.com/project-march/march/pull/404/files

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py [hunk 2]
```diff
@@ -1,34 +1,79 @@
...
+                                                     urdf_joint.limit.velocity),
+                                              default_setpoints,
+                                              duration
+                                              )
+            joint.gait_generator = gait_generator
+            joint_list.append(joint)
+        return cls(joint_list, duration, gait_type)
+
+    @classmethod
+    def gait_generator_from_file(cls, gait_generator, robot, filename):
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py]: Why did you call this `gait_generator_from_file`, can't you name it `from_file` to override it and then call `Subgait.from_file`?

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py [hunk 3]
```diff
@@ -23,6 +23,40 @@ def __init__(self, name, limits, setpoints, duration, gait_generator):
...
+                       limits,
+                       setpoints,
+                       duration
+                       )
+
+        rospy.logwarn('This subgait has no user defined setpoints.')
+        return cls.super(subgait_dict, joint_name, limits, duration)
+
+    @staticmethod
+    def get_setpoint_at_duration(joint_trajectory, joint_name, duration):
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py]: if this method is only used within this class you should name it `_get_setpoint_at_duration`

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py [hunk 4]
```diff
@@ -100,7 +136,10 @@ def remove_setpoint(self, index):
     def save_setpoints(self, single_joint_change=True):
         self.setpoints_history.append(copy.deepcopy(self.setpoints))    # list(...) to copy instead of pointer
         if single_joint_change:
-            self.gait_generator.save_changed_joints([self])
+            try:
+                self.gait_generator.save_changed_joints([self])
+            except AttributeError:
+                rospy.logerr("AttributeError: gait_generator not set in joint trajectories")
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_joint_trajectory.py]: Why did you add this `try`? I see you also removed the `gait_generator` attribute from the `__init__`, so where is it set now?

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py [hunk 5]
```diff
@@ -126,3 +171,33 @@ def get_mirror(self, key_1, key_2):
...
+
+    def set_duration(self, duration, rescale=False):
+        for joint in self.joints:
+            # Loop in reverse to avoid out of bounds errors while deleting.
+            for setpoint in reversed(joint.setpoints):
+                if rescale:
+                    setpoint.time = duration * setpoint.time / self.duration
+                else:
+                    if setpoint.time > duration:
+                        joint.setpoints.remove(setpoint)
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py]: You could also write this in a list comprehension
```python
joint.setpoints = [ s.time = duration * s.time / self.duration for s in joint.setpoints if s.time <= duration ]
```

Comment (Roelemans) [reply march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py]: Nope. That shit don't work. You can't assign a value in a list comprehension. Removing the loop would require some very fancy and annoying coding, which will not make this any clearer, so I will leave it like this.

File: march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py [hunk 6]
```diff
@@ -1,34 +1,77 @@
...
 
 class ModifiableSubgait(Subgait):
-    def to_joint_trajectory(self):
-        joint_trajectory = JointTrajectory()
+    joint_class = ModifiableJointTrajectory
+
+    @classmethod
+    def empty_subgait(cls, gait_generator, robot, gait_type='walk_like', duration=8):
+        if robot is None:
+            rospy.logerr("Cannot create gait without a loaded robot.")
```
Comment (Olavhaasie) [inline march_rqt_gait_generator/src/march_rqt_gait_generator/model/modifiable_subgait.py]: This if statement will only print an error and then continue
